### PR TITLE
Update equality tests and fixes errors.

### LINKED ISSUE #370: backend/models/test.js files are not using expect method correctly
### Overview
These test files need to use expect(VAL).toBe(VAL), which they are currently not doing.

### Action Items
REPLACE
    expect(savedData.name === submittedData.name);
WITH
    expect(savedData.name).toBe(submittedData.name);

@nickbeaird brought this up in a PR. I looked at all the test.js files for models, and they all were using expect incorrectly.

### LINKED ISSUE #417: Suppress temporarily the test causing GitHub actions failure
### Overview
The test https://github.com/hackforla/VRMS/blob/development/backend/config/auth.config.test.js fails on GitHub actions but passes locally. We need to suppress it or ignore programmatically, or just comment out. 

### Action Items

- [ ] Disable the test using any approach
- [ ] Add TODO to turn the test on after fixing GitHub actions.

### PR DESCRIPTION
**Pull request by awhig**

### CODE REVIEW

**Comment by awhig** [comment]
I'm not sure why but the test is not picking up the newly added dotenv-expand package, which is causing the test to fail.

**Comment by nickbeaird** [comment]
> I'm not sure why but the test is not picking up the newly added dotenv-expand package, which is causing the test to fail.



> I'm not sure why but the test is not picking up the newly added dotenv-expand package, which is causing the test to fail.

I am pretty sure that we have to explicitly add this to the secrets repo and the pipeline. Look at the Github Actions workflow, but we can pair on this at some point if you need help/access.

**Comment by awhig** [comment]
@nickbeaird Ok. I'm not familiar with the github action stuff or the secrets repo, so whenever you have time. I took a look at, and I think it does a yarn install, which I would think would pull down the package...

File: backend/globalConfig.json [hunk 1]
```diff
@@ -1 +1 @@
-{"mongoUri":"mongodb://127.0.0.1:42413/jest?","mongoDBName":"jest"}
\ No newline at end of file
+{"mongoUri":"mongodb://127.0.0.1:62109/jest?","mongoDBName":"jest"}
```
Comment (alex-anakin) [inline backend/globalConfig.json]: add an empty line at the file end

**Review by alex-anakin — CHANGES_REQUESTED** [review]
Looks good to me. Please resolve the comment and it will be good to go.
I believe we need to fix GitHub actions in a separate PR.

File: backend/app.js [hunk 2]
```diff
@@ -12,7 +12,11 @@ const customRequestHeaderName = 'x-customrequired-header';
 const dontCheckCustomRequestHeaderApis = ["GET::/api/recurringevents"];
 
 // Import environment variables
-require("dotenv").config();
+const dotenv = require('dotenv');
+const dotenvExpand = require('dotenv-expand');
```
Comment (nickbeaird) [inline backend/app.js:16]: 👍

File: backend/config/auth.config.test.js [hunk 3]
```diff
@@ -1,4 +1,4 @@
 test('Environment variables are working as expected', () => {
   const backendUrl = process.env.REACT_APP_PROXY;
-  expect(backendUrl === `http://localhost:${process.env.BACKEND_PORT}`);
+  expect(backendUrl).toEqual(`http://localhost:${process.env.BACKEND_PORT}`);
```
Comment (nickbeaird) [inline backend/config/auth.config.test.js]: I was looking at toEqual and it is used for comparing objects, and not strings. This above is comparing two strings through string interpolation. I am pretty sure that is toBe() syntax instead. Both may work, and you can test this by trying to write a toEqual test that you know should fail and to see if it meets your expectations. Then rerun with a toBe() and compare. 


https://jestjs.io/docs/en/expect#toequalvalue
https://jestjs.io/docs/en/expect#tobevalue

File: backend/models/checkin.test.js [hunk 4]
```diff
@@ -15,9 +15,9 @@ describe("Checkin Model saves the correct values", () => {
     await CheckIn.create(submittedCheckinData);
     const savedCheckinDataArray = await CheckIn.find();
     const savedCheckinData = savedCheckinDataArray[0];
-    expect(savedCheckinData.userId === submittedCheckinData.userId);
-    expect(savedCheckinData.eventId === submittedCheckinData.eventId);
-    expect(savedCheckinData.createdDate === submittedCheckinData.createdDate);
+    expect(savedCheckinData.userId).toEqual(submittedCheckinData.userId);
+    expect(savedCheckinData.eventId).toEqual(submittedCheckinData.eventId);
+    expect(savedCheckinData.createdDate.getTime()).toEqual(submittedCheckinData.createdDate);
```
Comment (nickbeaird) [inline backend/models/checkin.test.js]: I think that you can either compare the values with toBe(), or look at comparing the object with toEqual().

**Review by nickbeaird — CHANGES_REQUESTED** [review]

File: backend/models/checkin.test.js (hunk 4 above)
Comment (awhig) [reply backend/models/checkin.test.js]: In this case they are equivalent. I prefer to use toEqual as I don't really want to check object ref. equality.

File: backend/config/auth.config.test.js (hunk 3 above)
Comment (awhig) [reply backend/config/auth.config.test.js]: In this case you can use either toBe or toEqual. I prefer to use toEqual as I'm not checking for object ref. equality.

Comment (awhig) [reply backend/config/auth.config.test.js]: @nickbeaird I changed to toBe

File: backend/models/checkin.test.js (hunk 4 above)
Comment (awhig) [reply backend/models/checkin.test.js]: @nickbeaird I changed to toBe

**Pull request merged**